### ISSUE TITLE
[Question] Streaming Gaussian process regression

### ISSUE THREAD

**Issue by metab0t** [cornellius-gp/gpytorch#1823]
I want to do Gaussian process regression on a data stream where Gaussian process should be updated progressively and new predictions are made when new data comes.
I am aware of https://github.com/cornellius-gp/gpytorch/issues/1784. If I understand correctly, `get_fantasy_model` and `set_train_data` do different things.

- `get_fantasy_model` merges the provided input samples with existing samples
- `set_train_data` use the provided input samples to replace existing samples

For streaming Gaussian process regression, `get_fantasy_model` should be used. However, I wonder if the prediction performance will degrade when number of samples increases.

**Comment by wjmaddox** [comment]
Yes, the performance may degrade if the estimated hyperparameters slowly stop being a good fit for the response data. But, otherwise, the performance should improve.

**Comment by metab0t** [comment]
Thanks! Yes, stale hyperparameter may cause performance degradation.
I am also concerned about the speed of prediction. Will it be more and more slow to predict as size of samples increases? (because we need to inverse the covariance matrix of training samples)

**Comment by wjmaddox** [comment]
Yes, that'll in general slow down over time at a cost of `O(n)` _unless_ you're using [SKI](https://docs.gpytorch.ai/en/stable/examples/02_Scalable_Exact_GPs/KISSGP_Regression.html) (via its online version WISKI).

**Comment by metab0t** [comment]
I have seen your repo and paper: https://github.com/wjmaddox/online_gp
I will spend some time reading the paper. Thank you!

**Comment by wjmaddox** [comment]
Btw, this is implemented natively inside of GPyTorch now as part of the `get_fantasy_model` function for tradiational SKI models.
